osa/provenance/utils.py

```diff
@@ -29,31 +29,39 @@ def parse_variables(class_instance):
     # 02006.0000
     # LST1
 
-    configfile = cfg.get("lstchain", "dl1ab_config")
+    configfile_dl1 = cfg.get("lstchain", "dl1ab_config")
+    configfile_dl2 = cfg.get("lstchain", "dl2_config")
     rawdir = cfg.get("LST1", "R0_DIR")
-    r0_prefix = cfg.get("PATTERN", "R0PREFIX")
-    dl1_prefix = cfg.get("PATTERN", "DL1PREFIX")
-    dl2_prefix = cfg.get("PATTERN", "DL2PREFIX")
     rf_models_directory = cfg.get("lstchain", "RF_MODELS")
     calib_dir = cfg.get("LST1", "CALIB_DIR")
     dl1_dir = cfg.get("LST1", "DL1_DIR")
     dl2_dir = cfg.get("LST1", "DL2_DIR")
     nightdir = lstdate_to_dir(options.date)
+    analysis_dir = Path(cfg.get("LST1", "ANALYSIS_DIR")) / nightdir / options.prod_id
+    outdir_dl1 = Path(dl1_dir) / nightdir / options.prod_id / options.dl1_prod_id
+    outdir_dl2 = Path(dl2_dir) / nightdir / options.prod_id / options.dl2_prod_id
 
     if class_instance.__name__ == "r0_to_dl1":
-        # calibrationfile   [0] /fefs/aswg/data/real/calibration/20200218/v00/calibration.Run02006.0000.hdf5
-        # pedestalfile      [1] /fefs/aswg/data/real/calibration/20200218/v00/drs4_pedestal.Run02005.0000.fits
-        # time_calibration  [2] /fefs/aswg/data/real/calibration/20191124/v00/time_calibration.Run01625.0000.hdf5
-        # drivefile         [3] /fefs/home/lapp/DrivePositioning/drive_log_20_02_18.txt
-        # runsummary_file   [4] /fefs/aswg/data/real/monitoring/RunSummary/RunSummary_20200101.ecsv
+        # calibrationfile   [0] .../20200218/v00/calibration.Run02006.0000.hdf5
+        # pedestalfile      [1] .../20200218/v00/drs4_pedestal.Run02005.0000.fits
+        # time_calibration  [2] .../20191124/v00/time_calibration.Run01625.0000.hdf5
+        # drivefile         [3] .../DrivePositioning/drive_log_20_02_18.txt
+        # runsummary_file   [4] .../RunSummary/RunSummary_20200101.ecsv
         # run_str           [5] 02006.0000
-        # historyfile       [6] /fefs/aswg/data/real/running_analysis/20200218/v0.4.3_v00/sequence_LST1_02006.0000.history
 
-        class_instance.AnalysisConfigFile = configfile
+        class_instance.ObservationRun = class_instance.args[5].split(".")[0]
+        class_instance.ObservationDate = nightdir
+        class_instance.SoftwareVersion = options.lstchain_version
```

Comment:
maybe we could use `lstchain.__version__` to avoid the usage of `options`

Reply:
I would actually replace `options.lstchain_version` everywhere in  `lstosa` by `get_lstchain_version()` as it is already done in `osa/utils/utils.py`

Reply:
I agree. That function basically returns `f"v{lstchain.__version__}"`

I think that I implemented the options.lstchain_version to avoid importing lstchain several times, which takes several seconds. On the other hand, I want to get rid of the global `options` module at some point.